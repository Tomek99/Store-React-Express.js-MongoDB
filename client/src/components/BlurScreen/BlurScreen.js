import React from "react";
import styles from "./BlurScreen.module.scss";
import PropTypes from "prop-types";

function BlurScreen({
  isCartOpen,
  isSearchOpen,
  isNavigationOpen,
  handleAside,
  handleCart,
  handleNavigation,
  handleSearch,
}) {
  function handleBlurScreen() {
    if (isCartOpen) handleCart();
    else if (isNavigationOpen) handleNavigation();
    else if (isSearchOpen) handleSearch();
    else handleAside();
  }

  return <div className={styles.BlurScreen} onClick={handleBlurScreen}></div>;
}

BlurScreen.propTypes = {
  isCartOpen: PropTypes.bool,
  isSearchOpen: PropTypes.bool,
  isNavigationOpen: PropTypes.bool,
  handleAside: PropTypes.func,
  handleCart: PropTypes.func,
  handleNavigation: PropTypes.func,
  handleSearch: PropTypes.func,
};
export default BlurScreen;
