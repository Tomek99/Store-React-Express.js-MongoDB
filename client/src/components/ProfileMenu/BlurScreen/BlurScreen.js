import React from "react";
import styles from "./BlurScreen.module.scss";
import PropTypes from "prop-types";

function BlurScreen({ handleBlurScreen }) {
  return <div className={styles.BlurScreen} onClick={handleBlurScreen}></div>;
}
BlurScreen.propTypes = {
  handleBlurScreen: PropTypes.func,
};
export default BlurScreen;
