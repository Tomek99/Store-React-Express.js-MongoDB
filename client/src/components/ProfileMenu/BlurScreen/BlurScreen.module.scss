.BlurScreen {
    position: fixed;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    background: rgb(0, 0, 0);
    will-change: opacity;
    opacity: 0.64;
    inset: 0px;
    transition: opacity 300ms ease 0s;
    pointer-events: auto;
    z-index: 100;
}
