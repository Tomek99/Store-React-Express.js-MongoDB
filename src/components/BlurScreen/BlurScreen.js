import React from "react";

function BlurScreen() {
  return <div>BlurScreen</div>;
}

export default BlurScreen;
