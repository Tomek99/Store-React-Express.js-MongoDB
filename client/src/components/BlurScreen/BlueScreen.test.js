import { render, screen } from "@testing-library/react";
import BlurScreen from "./BlurScreen";

test("should first", () => {
  const { container } = render(<BlurScreen />);

  expect(container).toMatchSnapshot();
});
