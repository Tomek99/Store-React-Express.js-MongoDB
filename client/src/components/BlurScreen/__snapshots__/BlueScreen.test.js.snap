// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<div>
  <div
    class="BlurScreen"
  />
</div>
`;
